src/main/java/core/basesyntax/data/ReportGenerator.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ReportGenerator {
+    public List<String> generateReport(Map<String, Integer> inventory) {
+        List<String> result = new ArrayList<>();
+        result.add("fruit,quantity");
+        for (String key : inventory.keySet()) {
+            result.add(key + "," + inventory.get(key));
```

Comment:
To enhance readability and maintainability, consider using `Map.Entry<String, Integer>` in the for-each loop instead of iterating over the key set and then fetching the value. This will make the code cleaner and potentially more efficient.